### PR TITLE
feat(#1791): block height check

### LINKED ISSUE #1791: Update nodeHealth to use `x-block-height` header to check data node health
#2701 improved the way `NodeHealth` works, but misses one imporant part of the ticket. Data Node adds some metadata headers on every request:

```
< x-block-height: 926035
< x-block-timestamp: 1675424066275232000
< x-vega-connection: CONNECTED
< x-vega-node-id: n11.testnet.vega.xyz
```

These are interesting because they contain the data node's idea of the block height, timestamp, and whether or not it is connected to a core vega node. This is different to the `blockHeight` property in statistics, which comes directly from the  core node.

To put it another way, if you run `curl -v https://api.n11.testnet.vega.xyz/statistics`, the response header `x-block-height` could contain a different height to the body property `.statistics.blockHeight`. It this is the case, it is an indicator that the data you are being served is not up to date, which is a very important thing to flag up to the user.

Screenshot 2023-02-03 at 12 08 41

## Acceptance Criteria

- [ ] As a user I can clearly see that the app I am on is connected to a datanode that isn’t up to date 

## Tasks

- [ ] Update `Nodehealth` component to log `x-block-height` for the data node you are connected to
- [ ] For the data node you are connected to, compare `x-block-height` to the currently held block height value (i.e. the one read from statistics)
- [ ] Use the currently acceptable value of `BLOCK_THRESHOLD` - if `x-block-height` is more than `BLOCK_THRESHOLD` *behind* `statistics.blockHeight`, show an error indicating stale data

### PR DESCRIPTION
**Pull request by mattrussell36**
# Related issues 🔗

Closes #1791 

# Description ℹ️

Checks a random subset of nodes every 30s to see if the your current nodes block height is up to date. This is shown in the UI by an indicator in the footer along with the text Opertaional, Non operational, X blocks behind or Offline

# Demo 📺

drawing

drawing

drawing

drawing

# Technical
- Updated to createClient to take a few more options, for retry and connect to dev tools. This tidies up the network quite a bit when attempting connections and also doesn't mess with Apollo dev tools
- Fixed broken/evergreen tests for environment provider. Looks like this has been broken since [this commit](https://github.com/vegaprotocol/frontend-monorepo/commit/71ede25339e7a5cd2c8acd8198d99e72244e9a02#diff-771a20a1909c122b0792e947e22ec89f81061f1b41782d1bf7324536b464d14a)
- Adds useNavigatorOnline hook to detect if browser is connected or not

### CODE REVIEW

File: libs/environment/src/hooks/use-node-health.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+// Number of nodes to query against
+export const NODE_SUBSET_COUNT = 5;
+
+// Queries all nodes from the environment provider via an interval
+// to calculate and return the difference between the most advanced block
+// and the block height of the current node
+export const useNodeHealth = (clients: ClientCollection, vegaUrl?: string) => {
+  const [blockDiff, setBlockDiff] = useState(0);
+
+  useEffect(() => {
```
Comment (asiaznik) [inline libs/environment/src/hooks/use-node-health.ts:20]: suggestion (non-blocking): I'm a little bit tempted to refactor this into `useSyncExternalStore` as the node health function is to literally sync with the block heights of other nodes. What do you think?

File: apps/trading/components/footer/footer.tsx [hunk 2]
```diff
@@ -1,28 +1,70 @@
...
+const BLOCK_THRESHOLD = 3;
+
+export const NodeHealth = ({
+  blockDiff,
+  openNodeSwitcher,
+}: NodeHealthProps) => {
+  let intent = Intent.Success;
+  let text = 'Operational';
+
+  if (blockDiff < 0) {
```
Comment (asiaznik) [inline apps/trading/components/footer/footer.tsx:55]: question (non-blocking): How about a simple online/offline (browser, node) check here as well?

File: libs/environment/src/hooks/use-node-health.ts (hunk 1 above)
Comment (mattrussell36) [reply libs/environment/src/hooks/use-node-health.ts:20]: As discussed, the store is simply normal react state, so using useSyncExternalStore here doesn't provide much benefit

File: apps/trading/components/footer/footer.tsx (hunk 2 above)
Comment (mattrussell36) [reply apps/trading/components/footer/footer.tsx:55]: Ill add a check and render 'Offline'

**Review by asiaznik — APPROVED** [review]

**Pull request merged**